### DISCUSSION TITLE
Multiple grok patterns in fluentd

### DISCUSSION THREAD

**Discussion by vikasmishra17** [fluent/fluentd#3665] [Q&A]
I am facing a problem while using fluentd-1.14.3. The logs are not being parsed even when I am using the grok pattern which ran properly in Logstash. 

Does the fluent-plugin-grok-parser support multi-format logs from a single file.

I am tailing single haproxy log files. Each of them may have a different format.

Is it possible to use the grok parser in this situation? Help is appreciated

**Comment by fujimotos** [comment]
It's hard to tell what you're trying to do without additional info

The probability is that you need to use `parser_multiline` with
`in_tail` to handle events that spans across multiple lines:

https://docs.fluentd.org/parser/multiline

or use `fluent-plugin-concat` to concatinate them manually:

https://github.com/fluent-plugins-nursery/fluent-plugin-concat
